components/apimgt/org.wso2.carbon.apimgt.eventing.hub/src/main/java/org/wso2/carbon/apimgt/eventing/hub/EventHubEventPublisherFactory.java

```diff
@@ -0,0 +1,72 @@
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.eventing.EventPublisher;
+import org.wso2.carbon.apimgt.eventing.EventPublisherFactory;
+import org.wso2.carbon.apimgt.eventing.EventPublisherType;
+import org.wso2.carbon.apimgt.eventing.hub.internal.ServiceReferenceHolder;
+import org.wso2.carbon.event.output.adapter.core.OutputEventAdapterConfiguration;
+import org.wso2.carbon.event.output.adapter.core.exception.OutputEventAdapterException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Factory for getting event hub event publisher instances.
+ */
+public class EventHubEventPublisherFactory implements EventPublisherFactory {
+    private static final Log log = LogFactory.getLog(EventHubEventPublisherFactory.class);
+
+    @Override
+    public void configure(Map<String, String> configuration) {
+        OutputEventAdapterConfiguration adapterConfiguration = new OutputEventAdapterConfiguration();
+        adapterConfiguration.setName(EventHubEventPublisherConstants.EVENT_HUB_NOTIFICATION_EVENT_PUBLISHER);
+        adapterConfiguration.setType(EventHubEventPublisherConstants.BLOCKING_EVENT_TYPE);
+        adapterConfiguration.setMessageFormat(EventHubEventPublisherConstants.BLOCKING_EVENT_FORMAT);
+        Map<String, String> adapterParameters = new HashMap<>();
+        adapterParameters.put(EventHubEventPublisherConstants.RECEIVER_URL,
+                configuration.get(EventHubEventPublisherConstants.RECEIVER_URL));
+        adapterParameters.put(EventHubEventPublisherConstants.AUTHENTICATOR_URL,
+                configuration.get(EventHubEventPublisherConstants.AUTHENTICATOR_URL));
+        adapterParameters.put(EventHubEventPublisherConstants.USERNAME,
+                configuration.get(EventHubEventPublisherConstants.USERNAME));
+        adapterParameters.put(EventHubEventPublisherConstants.PASSWORD,
+                configuration.get(EventHubEventPublisherConstants.PASSWORD));
+        adapterParameters.put(EventHubEventPublisherConstants.PROTOCOL,
+                configuration.get(EventHubEventPublisherConstants.PROTOCOL));
+        adapterParameters.put(EventHubEventPublisherConstants.PUBLISHING_MODE,
+                configuration.get(EventHubEventPublisherConstants.PUBLISHING_MODE));
+        adapterParameters.put(EventHubEventPublisherConstants.PUBLISHING_TIME_OUT,
+                configuration.get(EventHubEventPublisherConstants.PUBLISHING_TIME_OUT));
+        adapterConfiguration.setStaticProperties(adapterParameters);
+        try {
+            ServiceReferenceHolder.getInstance().getOutputEventAdapterService().create(adapterConfiguration);
+        } catch (OutputEventAdapterException e) {
+            log.warn("Exception occurred while creating WSO2 Event Adapter. Event notification may not work "
+                    + "properly", e);
+        }
+    }
+
+    @Override
+    public EventPublisher getEventPublisher(EventPublisherType eventPublisherType) {
```

Comment:
How are we going to use the types here?

Reply:
A publisher is retrived by the type. It is not required at the moment for the hub due to the way it is written. But we need it for the other implementation. + If we remove the thrift listener in the future and write the jms publisher we will need it.